californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -840,39 +840,55 @@ private Request ifNoneMatch(Request request) {
 		return request;
 	}
 	
-	/*
+	/**
 	 * Sends the specified observe request and waits for the response whereupon
 	 * the specified handler is invoked when a notification arrives.
 	 *
 	 * @param request the request
+	 * 
 	 * @param handler the Response handler
+	 * 
 	 * @return the CoAP observe relation
-	 */
-	private CoapObserveRelation observeAndWait(Request request, CoapHandler handler) {
-		Endpoint outEndpoint = getEffectiveEndpoint(request);
-		CoapObserveRelation relation = new CoapObserveRelation(request, outEndpoint);
-		request.addMessageObserver(new ObserveMessageObserverImpl(handler, relation));
-		CoapResponse response = synchronous(request, outEndpoint);
-		if (response == null || !response.advanced().getOptions().hasObserve())
-			relation.setCanceled(true);
-		relation.setCurrent(response);
-		return relation;
+	 * @throws IllegalStateException if the observe option is not set in the
+	 *             request
+	 */
+	public CoapObserveRelation observeAndWait(Request request, CoapHandler handler) {
+		if (request.getOptions().hasObserve()) {
+			Endpoint outEndpoint = getEffectiveEndpoint(request);
+			CoapObserveRelation relation = new CoapObserveRelation(request, outEndpoint);
+			request.addMessageObserver(new ObserveMessageObserverImpl(handler, relation));
+			CoapResponse response = synchronous(request, outEndpoint);
+			if (response == null || !response.advanced().getOptions().hasObserve())
+				relation.setCanceled(true);
+			relation.setCurrent(response);
+			return relation;
+		} else {
+			throw new IllegalStateException("please make sure that the request has observe option set.");
```

Comment:
since this is a problem with the provided parameter value (`request`), we should better throw an `IllegalArgumentException'